proxy/handler_test.go

```diff
@@ -0,0 +1,68 @@
+package proxy
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHandler(t *testing.T) {
+	expectedRespBody := "OK"
```

Comment:
This test is a unit test for handler without its dependencies. It only tests whether all the required functions are called (`SignRequest` and metric stuff) without actually calling anything.

Reply:
Thank you for the explanation. But still I think it would make sense to at least split this test into some private functions / steps to better reflect the intention.

Reply:
I don't quite get it, can you elaborate on why and how?

Reply:
why - to improve readability
how - you could make it something like (note it's just an example)
```
prepareFakeEndpoint(url)
createHandler()
response := serveTheRequest()
verifyStatusAndBody(response)
```

Reply:
ah ok, got it. I thought u meant changing the test logic into multiple tests. Imma find a way to improve that